Tests/Shared.Specs/TestingTimer.cs

```diff
@@ -6,31 +6,54 @@
 namespace FluentAssertions.Specs
 {
     /// <summary>
-    /// Implementation of <see cref="ITimer"/> for testing purposes only.
+    /// Implementation of <see cref="IClock"/> for testing purposes only.
     /// </summary>
-    internal class TestingTimer : ITimer
+    internal class TestClock : IClock
     {
-        private TaskCompletionSource<bool> Signal { get; } = new TaskCompletionSource<bool>();
+        private DateTime now = DateTime.Now;
```

Comment:
The `now` is quite confusing to me. The time will be changed with each invocation of `Delay`.
You should add some comments to explain the complex behavior.

Reply:
I added some comments.